src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -795,23 +795,26 @@ public Task<List<RawFunctionMetadata>> GetFunctionMetadata()
 
         internal Task<List<RawFunctionMetadata>> SendFunctionMetadataRequest()
         {
-            // reset indexing task when in case we need to send another request
-            _functionsIndexingTask = new TaskCompletionSource<List<RawFunctionMetadata>>(TaskCreationOptions.RunContinuationsAsynchronously);
+            lock (_syncLock)
```

Comment:
Do we want to reuse the existing lock here? Is this synchronization with the other paths using this required? Have we reviewed the other usages to avoid synchronization/lock issues?

Reply:
Discussed additional changes offline as the task would be overridden with multiple calls.

Reply:
Have fixed this